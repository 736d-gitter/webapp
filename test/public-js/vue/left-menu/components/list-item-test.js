'use strict';

const { createLocalVue, shallowMount } = require('@vue/test-utils');
const Vuex = require('vuex');

const createStore = require('../../../../../public/js/vue/store').default;
const actions = require('../../../../../public/js/vue/store/actions');
const ListItem = require('../../../../../public/js/vue/left-menu/components/list-item.vue');

const {
  createSerializedRoomFixture,
  createSerializedOneToOneRoomFixture
} = require('../../fixture-helpers');

let wrapper;
let stubbedActions = {};
function factory(propsData = {}, extendStore = () => {}) {
  const localVue = createLocalVue();
  localVue.use(Vuex);

  Object.keys(actions).forEach(actionKey => {
    stubbedActions[actionKey] = jest.fn();
  });

  const store = createStore({
    actions: stubbedActions
  });
  extendStore(store);

  wrapper = shallowMount(ListItem.default, {
    localVue,
    store,
    propsData
  });
}

describe('list-item', () => {
  afterEach(() => {
    wrapper.destroy();
  });

  it('community room matches snapshot', () => {
    factory({
      item: createSerializedRoomFixture('my-community/community')
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('room with short name matches snapshot', () => {
    factory({
      item: createSerializedRoomFixture('foo/bar')
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('room with long name matches snapshot', () => {
    factory({
      item: createSerializedRoomFixture('abcdefghijklmnop/qrstuvwxyz')
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('one to one room matches snapshot', () => {
    factory({
      item: createSerializedOneToOneRoomFixture('EricGitterTester')
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('favourite room matches snapshot', () => {
    factory({
      item: {
        ...createSerializedRoomFixture('my-community/community'),
        favourite: 1
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('favourite loading room matches snapshot', () => {
    factory({
      item: {
        ...createSerializedRoomFixture('my-community/community'),
        favourite: 1,
        loading: true
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('calls store action "changeDisplayedRoom" and "toggleLeftMenu" when item is clicked', () => {
    const room = createSerializedRoomFixture('my-community/community');
    factory({
      item: room
    });

    wrapper.find({ ref: 'link' }).trigger('click');

    expect(stubbedActions.changeDisplayedRoom).toHaveBeenCalledWith(
      expect.anything(),
      room.id,
      undefined
    );

    expect(stubbedActions.toggleLeftMenu).toHaveBeenCalledWith(expect.anything(), false, undefined);
  });
});
