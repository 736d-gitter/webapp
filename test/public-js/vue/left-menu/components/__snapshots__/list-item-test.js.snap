// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list-item active room matches snapshot 1`] = `
<li
  class="list-item"
>
  <a
    class="list-item-link is-active"
    href="http://localhost:5000/my-community/room1"
  >
    <img
      class="list-item-avatar"
      src="http://localhost:5000/api/private/avatars/group/i/5a8739841543b98772a686a8"
    />
     
    <h2
      class="list-item-name"
    >
      <span
        class="name-piece"
      >
        room1
      </span>
    </h2>
     
    <!---->
     
    <!---->
  </a>
</li>
`;

exports[`list-item community room matches snapshot 1`] = `
<li
  class="list-item"
>
  <a
    class="list-item-link"
    href="http://localhost:5000/my-community/community"
  >
    <img
      class="list-item-avatar"
      src="http://localhost:5000/api/private/avatars/group/i/5a8739841543b98772a686a8"
    />
     
    <h2
      class="list-item-name"
    >
      <span
        class="name-piece"
      >
        my-community
      </span>
    </h2>
     
    <!---->
     
    <!---->
  </a>
</li>
`;

exports[`list-item favourite loading room matches snapshot 1`] = `
<li
  class="list-item is-favourited"
  data-favourite="1"
>
  <a
    class="list-item-link"
    href="http://localhost:5000/my-community/community"
  >
    <img
      class="list-item-avatar"
      src="http://localhost:5000/api/private/avatars/group/i/5a8739841543b98772a686a8"
    />
     
    <h2
      class="list-item-name"
    >
      <span
        class="name-piece"
      >
        my-community
      </span>
    </h2>
     
    <loading-spinner-stub />
     
    <!---->
  </a>
</li>
`;

exports[`list-item favourite room matches snapshot 1`] = `
<li
  class="list-item is-favourited"
  data-favourite="1"
>
  <a
    class="list-item-link"
    href="http://localhost:5000/my-community/community"
  >
    <img
      class="list-item-avatar"
      src="http://localhost:5000/api/private/avatars/group/i/5a8739841543b98772a686a8"
    />
     
    <h2
      class="list-item-name"
    >
      <span
        class="name-piece"
      >
        my-community
      </span>
    </h2>
     
    <!---->
     
    <!---->
  </a>
</li>
`;

exports[`list-item one to one room matches snapshot 1`] = `
<li
  class="list-item"
>
  <a
    class="list-item-link"
    href="http://localhost:5000/EricGitterTester"
  >
    <img
      class="list-item-avatar"
      src="http://localhost:5000/api/private/avatars/gh/uv/4/EricGitterTester"
    />
     
    <h2
      class="list-item-name"
    >
      <span
        class="name-piece"
      >
        EricGitterTester
      </span>
    </h2>
     
    <!---->
     
    <!---->
  </a>
</li>
`;

exports[`list-item room with long name matches snapshot 1`] = `
<li
  class="list-item"
>
  <a
    class="list-item-link"
    href="http://localhost:5000/abcdefghijklmnop/qrstuvwxyz"
  >
    <img
      class="list-item-avatar"
      src="http://localhost:5000/api/private/avatars/group/i/5a8739841543b98772a686a8"
    />
     
    <h2
      class="list-item-name"
    >
      <span
        class="name-piece"
      >
        qrstuvwxyz
      </span>
    </h2>
     
    <!---->
     
    <!---->
  </a>
</li>
`;

exports[`list-item room with short name matches snapshot 1`] = `
<li
  class="list-item"
>
  <a
    class="list-item-link"
    href="http://localhost:5000/foo/bar"
  >
    <img
      class="list-item-avatar"
      src="http://localhost:5000/api/private/avatars/group/i/5a8739841543b98772a686a8"
    />
     
    <h2
      class="list-item-name"
    >
      <span
        class="name-piece"
      >
        foo/bar
      </span>
    </h2>
     
    <!---->
     
    <!---->
  </a>
</li>
`;
